packages/did/src/Did.chain.ts

```diff
@@ -419,19 +441,172 @@ export async function getStoreTx(
     .createType(api.tx.did.create.meta.args[0].type.toString(), apiInput)
     .toU8a()
 
-  const signature = await sign({
+  const { signature } = await sign({
     data: encoded,
-    keyRelationship: 'authentication',
+    verificationRelationship: 'authentication',
   })
   const encodedSignature = {
-    [signature.keyType]: signature.signature,
+    [authenticationKey.type]: signature,
   } as EncodedSignature
   return api.tx.did.create(encoded, encodedSignature)
 }
 
+/**
+ * Create a DID creation operation which would write to chain the DID Document provided as input.
+ * Only the first authentication, assertion, and capability delegation verification methods are considered from the input DID Document.
+ * All the input DID Document key agreement verification methods are considered.
+ *
+ * The resulting extrinsic can be submitted to create an on-chain DID that has the provided verification methods and services.
+ *
+ * A DID creation operation can contain at most 25 new services.
+ * Additionally, each service must respect the following conditions:
+ * - The service ID is at most 50 bytes long and is a valid URI fragment according to RFC#3986.
+ * - The service has at most 1 service type, with a value that is at most 50 bytes long.
+ * - The service has at most 1 URI, with a value that is at most 200 bytes long, and which is a valid URI according to RFC#3986.
+ *
+ * @param input The DID Document to store.
+ * @param submitter The KILT address authorized to submit the creation operation.
+ * @param sign The sign callback. The authentication key has to be used.
+ *
+ * @returns The SubmittableExtrinsic for the DID creation operation.
+ */
+export async function getStoreTxFromDidDocument(
```

Comment:
I just split up the old function that was taking either a DID Document or a bunch of keys and services. Now instead of one function that takes both, it's two functions that each take one, with a clearer naming. What would you suggest to do here?

Reply:
Fair point. I noticed it was only used in tests, so I moved it in the `TestUtils` module: https://github.com/KILTprotocol/sdk-js/pull/799/commits/6e42e192e71f2acbdd0b4aec2f6a3c7d8cdef535.